blocks/collapsible/collapsible.js

```diff
@@ -0,0 +1,83 @@
+export default function decorate(block) {
```

Comment:
this can be shortened considerably, by making use of the `add` and `toggle` methods provided by an element's `classList`.

I would remove all of the transition and style effects and replace them with CSS, effectively only using JS for where it is absolutely necessary.

JS:
```
export default function decorate(block) {
  const processRow = (row) => {
    const [questionArea, answerArea] = Array.from(row.children);
    const question = document.createElement('a');
    const answerBlock = document.createElement('p');

    questionArea.classList.add('question');
    answerArea.classList.add('answer', 'collapsed');

    question.textContent = questionArea.textContent;
    answerBlock.textContent = answerArea.textContent;

    questionArea.textContent = '';
    answerArea.textContent = '';

    questionArea.appendChild(question);
    answerArea.appendChild(answerBlock);

    const toggleAnswer = () => {
      answerArea.classList.toggle('collapsed');
      questionArea.classList.toggle('expanded');
    };

    questionArea.addEventListener('click', toggleAnswer);
  };

  Array.from(block.children).forEach(processRow);
};
```

CSS (just an example, adjust as necessary for actual visual representation):
```
.answer, .question {
  transition: all 0.5s ease-in-out;
  overflow: hidden;
}

.question {
  cursor: pointer;
}

.answer {
  height: 0;
  padding-top: 0;
  padding-bottom: 0;
}

.question.expanded + .answer {
  height: auto;
  padding-top: 1em;
  padding-bottom: 1em;
}
```

cc @Acidham

Reply:
@smiroslav in the code i provided on slack:
```
export default function decorate(block) {
  const processRow = (row) => {
    const [questionArea, answerArea] = Array.from(row.children);

    questionArea.classList.add('question');
    answerArea.classList.add('answer');
    answerArea.style.height = '0';

    questionArea.addEventListener('click', () => {
      questionArea.classList.toggle('expanded');
      if (questionArea.classList.contains('expanded')) {
        // Expand the answer
        answerArea.style.height = `${answerArea.scrollHeight}px`;
      } else {
        // Collapse the answer
        answerArea.style.height = '0';
      }
    });
  };

  Array.from(block.children).forEach(processRow);
}
```
CSS:
```
.answer, .question {
  overflow: hidden;
}

.question {
  cursor: pointer;
}

.answer {
  transition: height 0.5s ease-in-out;
}
```

The scrollHeight property should work correctly even if the element is initially hidden (via display: none, for example). However, there are certain circumstances where this might not be the case, such as if the parent of the hidden element has no dimensions or if CSS transitions are used that interfere with the calculation of scrollHeight.

Did you find this to be the case? I am curious to learn about it.

Reply:
@solaris007 I gave it another try and simplified the code. In addition to what you suggested, I set top and bottom padding when collapsed so we have transition for it as well 
```
.collapsible .answer.collapsed {
    height: 0;
    padding-top: 0;
    padding-bottom: 0;
} 
```
and in the listener added 
```
answerArea.classList.toggle('collapsed');
```
This time magically, it works. I guess in my previous tries, it did not work since starting point was ```display:none```.